### ISSUE TITLE
Eigenfaces with NN doesn't work

### ISSUE THREAD

**Issue by mkosik** [Qualeams/Android-Face-Recognition-with-Deep-Learning-Test-Framework#29]
When trying to run a "Recognition Training" for Eigenfaces with NN, Training Activity always stucks on the last trained image, trying to save the results, exhausts all the available memory & crashes.

The same happens for an app compiled from this repo with v1.5.3 lib (imported as a module, due to an arr corrupt problem) as well as currently published version on PlayStore.

I hope the following stacktrace will help:
AndroidRuntime: FATAL EXCEPTION: Thread-4914
            Process: ch.zhaw.facerecognition, PID: 18790
            Theme: themes:{com.android.settings=overlay:system, default=overlay:system, iconPack:system, fontPkg:system, com.android.systemui=overlay:system, com.android.systemui.navbar=overlay:system}
            java.lang.OutOfMemoryError: Failed to allocate a 85925328 byte allocation with 16777216 free bytes and 77MB until OOM
            at java.lang.AbstractStringBuilder.enlargeBuffer(AbstractStringBuilder.java:95)
            at java.lang.AbstractStringBuilder.append0(AbstractStringBuilder.java:146)
            at java.lang.StringBuilder.append(StringBuilder.java:216)
            at ch.zhaw.facerecognitionlibrary.Helpers.MatXml.dataStringBuilder(MatXml.java:283)
            at ch.zhaw.facerecognitionlibrary.Helpers.MatXml.writeMat(MatXml.java:258)
            at ch.zhaw.facerecognitionlibrary.Helpers.FileHelper.saveMatListToXml(FileHelper.java:129)
            at ch.zhaw.facerecognitionlibrary.Recognition.Eigenfaces.saveToFile(Eigenfaces.java:132)
            at ch.zhaw.facerecognitionlibrary.Recognition.Eigenfaces.train(Eigenfaces.java:68)
            at ch.zhaw.facerecognition.Activities.TrainingActivity$1.run(TrainingActivity.java:143)
            at java.lang.Thread.run(Thread.java:818)

**Comment by sladomic** [comment]
This is due to the bad implementation of the serialization (as mentioned here https://github.com/Qualeams/Android-Face-Recognition-with-Deep-Learning-Library/issues/16#issuecomment-327738283)

Until this has not been fixed, you will need to use workarounds, or implement a better serialization/OpenCV storage.

Workarounds are:

- use a smaller N value (e.g. 25)
- use less images
- use a smaller PCA threshold

Basically you need to decrease the matrix sizes
